### DISCUSSION TITLE
Restricting follow operations / follow limits

### DISCUSSION THREAD

**Discussion by tibor** [bluesky-social/atproto#3165] [General Discussion]
I see a lot of accounts with low post numbers and high follow numbers, i.e. tens of thousands of follows.

I think, this is the typical Follow-Spam like it occured on Twitter and other platforms as well.

[Twitter had some follow limits](https://help.x.com/en/using-x/x-follow-limit) active to limit follow spam:
* Every X account can follow up to 5,000 accounts. Once you reach that number, you may need to wait until your account has more followers before you can follow additional accounts. 
* This number is different for each account and is automatically calculated based on your unique ratio of followers to following.

What do you think, should we implement something similar in AtProto as well?

**Comment by CompThing** [comment]
I think a mechanism for restricting follower spam at a platform level would be useful and higher priority than these alternatives:

- manual reporting of follow-spam, follow back harvesting
- blocking of these accounts from a follower management UI mentioned in https://github.com/bluesky-social/social-app/issues/6449
As a user, I would like a mechanism that allows me to block follow-spam accounts. e.g. block following by accounts that are following 5000+ other accounts, or have a ratio of following vs follows of 100 or more.
The desirability of such a feature is probably univisersal leading to the feature being built in for all users
- -
